File: Java_Programmers/1단계/src/자릿수더하기.java
import java.util.Arrays;

public class 자릿수더하기 {
    public int solution(int n) {
        int answer = 0;
        String[] arr = String.valueOf(n).split("");
        for (String s : arr) {
            answer += Integer.valueOf(s);
        }
        return answer;
    }

    public int solution2(int n){
        int answer = 0;
        while (n != 0) {
            answer += n % 10;
            n /= 10;
        }
        return answer;
    }

    public static void main(String[] args) {
        자릿수더하기 sol = new 자릿수더하기();
        int n = 987;
        System.out.println(sol.solution(n));
        System.out.println(sol.solution2(n));
    }
}

File: python/python_Programmers/1단계/자릿수더하기.py
# 내 풀이
def solution(n):
    answer = 0
    for i in str(n):
        answer += int(i)
    return answer

# map을 이용한 풀이
def solution2(number):
    return sum(map(int,str(number)))

# 재귀를 이용한 풀이
def solution3(number):
    if number < 10:
        return number;
    return (number % 10) + solution3(number // 10)
    # number % 10은 1의자리를 구함
    # number // 10은 1의자리를 제거한 수를 다시 재귀로 1의자리구함
    # 1의자리만 남으면 그대로 리턴

n = 987
# print(solution(n))
print(solution2(n))